### PR TITLE
Release 1.3.0 version

### PR DESCRIPTION
**Pull request by MarceloSalazar**
Summary:
- Add callbacks for registered/unregistered, bring back update hub
- Add wrappers for PUT/POST/Notification callbacks
- Bring back memory_tests.cpp so it compiles with HEAP_STATS enabled
- Use m2m::String instead of std::string
- Allow overwriting default firmware path
- Remove app shield references from update_ui
- Update to Mbed Cloud Client public release 1.3.0

### CODE REVIEW

**Comment by MarceloSalazar** [comment]
@janjongboom can you have a look?
https://github.com/ARMmbed/simple-mbed-cloud-client/blob/2896e7fa8a29518b121e5a7580f8327f2699c2c7/simple-mbed-cloud-client.cpp#L271
https://github.com/ARMmbed/simple-mbed-cloud-client/blob/2896e7fa8a29518b121e5a7580f8327f2699c2c7/simple-mbed-cloud-client.cpp#L274

**Comment by MarceloSalazar** [comment]
I've added basic intro and how to use it.
Pending to clean up TODO before releasing.

**Comment by janjongboom** [comment]
@MarceloSalazar I've added a guard against calling `register_and_connect` multiple times. Not really sure how to clean up the rest, @bridadan wrote it and it seems to work.

File: simple-mbed-cloud-client.cpp [hunk 1]
```diff
@@ -267,29 +304,42 @@ bool SimpleMbedCloudClient::is_register_called() {
...
-                     resourceDef.put_callback, resourceDef.post_callback, resourceDef.notification_callback);
-        _resources[i]->set_resource(res);
+                    resourceDef.resource_id, resourceDef.name.c_str(), M2MResourceInstance::STRING,
+                    (M2MBase::Operation)resourceDef.method_mask, resourceDef.value.c_str(), resourceDef.observable,
+                    resourceDef.put_callback, resourceDef.post_callback, resourceDef.notification_callback);
+        _resources[i]->set_m2m_resource(res);
     }
     _cloud_client.add_objects(_obj_list);
 
+    _register_and_connect_called = true;
```
Comment (bridadan) [inline simple-mbed-cloud-client.cpp]: I doesn't look like we're setting this back to `false` when we deregister. Is the intent that this function should only be called once?

Comment (janjongboom) [reply simple-mbed-cloud-client.cpp]: Yes, as it adds new resources. You can call `call_register()` when doing it the second time. It's public and does not have this guard.

**Pull request merged**